include/rcpputils/filesystem_helper.hpp

```diff
@@ -183,16 +183,31 @@ inline bool create_directories(const path & p)
 {
   path p_built;
 
-  for (auto it = p.cbegin(); it != p.cend(); ++it) {
-    p_built /= *it;
-
+  int status = 0;
+  for (auto it = p.cbegin(); it != p.cend() && status == 0; ++it) {
+    if (p_built.empty()) {
+      p_built = *it;
+    } else {
+      p_built /= *it;
+    }
+    if (!p_built.exists()) {
 #ifdef _WIN32
-    _mkdir(p_built.string().c_str());
+      status = _mkdir(p_built.string().c_str());
 #else
-    mkdir(p_built.string().c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
+      status = mkdir(p_built.string().c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
```

Comment:
According to [create_directory docs](https://en.cppreference.com/w/cpp/filesystem/create_directory), the directory is created with `0777` perms (see [here](http://en.cppreference.com/w/cpp/filesystem/perms)).

Reply:
Also, shouldn't break the loop if status is not zero?

Reply:
> According to `create_directory` docs, the directory is created with 0777 perms (see here).

True. I do not know why the original author made it `0775` instead though.

> Also, shouldn't break the loop if status is not zero?

 It is.

Reply:
> It is.

Didn't realize.